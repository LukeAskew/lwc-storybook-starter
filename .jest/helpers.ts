import { render, TemplateResult } from "lit-html";

try {
  if ("afterEach" in window) {
    afterEach(() => {
      while (document.body.firstChild) {
        document.body.removeChild(document.body.firstChild);
      }
    });
  }
} catch (error) {
  console.error(error as string);
}

/**
 * Renders a template result into the document body
 */
export function fixture(template: TemplateResult) {
  const element = document.createElement("div");
  document.body.append(element);
  render(template, element);
  return element.children[0] as HTMLElement;
}

/**
 * Waits until the next event loop tick
 */
export function nextTick() {
  return new Promise((resolve) => {
    process.nextTick(resolve);
  });
}

/**
 * Waits until the current repaint is finished
 */
export function nextFrame() {
  return new Promise((resolve) => requestAnimationFrame(resolve));
}

/**
 * Waits for a number of milliseconds
 */
export function wait(ms: number = 200) {
  return new Promise((resolve) => {
    setTimeout(resolve, ms);
  });
}
