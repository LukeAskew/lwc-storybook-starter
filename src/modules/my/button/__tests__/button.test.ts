import { html } from "lit-html";
import { fixture } from "~/.jest/helpers";

import Button from "my/button";

customElements.define("my-button", Button.CustomElementConstructor);

describe("my-button", () => {
  it("works", () => {
    const element = fixture(html` <my-button>Test</my-button> `);
    expect(element).toMatchSnapshot();
  });
});
