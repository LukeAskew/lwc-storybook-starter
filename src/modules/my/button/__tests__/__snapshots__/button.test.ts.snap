// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my-button works 1`] = `
<my-button>
  #shadow-root(open)
    Test
    <button>
      <slot />
    </button>
</my-button>
`;
